### PR TITLE
feat: add <, <=, >, >= comparison operators to conditions

### PR DESCRIPTION
**Pull request by scottgl9**
## Summary
- Adds `<`, `<=`, `>`, `>=` to the workflow condition parser
- Enables numeric comparisons in `when:`/`condition:` expressions: `$step.json.count > 0`
- Consistent with the pipeline `where` command which already supports these operators
- Non-numeric comparisons safely return false (no crash)

## Example
```yaml
steps:
  - id: data
    run: gh pr list --json number --limit 100
  - id: alert
    run: curl -X POST https://slack.example.com/webhook
    when: $data.json | length > 50
  - id: range_check
    run: echo "in range"
    when: $data.json.score >= 1 && $data.json.score <= 100
```

## Changes
- `src/workflows/file.ts` — extended `ConditionToken` type, added `lt`/`lte`/`gt`/`gte` tokenization and evaluation with `numericCompare` helper
- `test/condition_comparison.test.ts` — 8 tests covering all operators, boundary conditions, combined expressions, NaN handling, and backward compatibility

## Test plan
- [x] All 8 new tests pass
- [x] All 11 existing workflow_file tests still pass
- [x] Build succeeds with no type errors

### CODE REVIEW

File: src/workflows/file.ts [hunk 1]
```diff
@@ -1193,6 +1205,13 @@ function compareConditionValues(left: unknown, right: unknown) {
   return Object.is(left, right);
 }
 
+function numericCompare(left: unknown, right: unknown, cmp: (a: number, b: number) => boolean): boolean {
+  const a = Number(left);
+  const b = Number(right);
+  if (Number.isNaN(a) || Number.isNaN(b)) return false;
```
[the remark this answers is not in the document]
Comment (scottgl9) [reply src/workflows/file.ts]: Fixed in 26f20d3 — `numericCompare` now uses a strict `isStrictlyNumeric()` guard that only accepts actual numbers and numeric strings. `true`, `false`, `null`, empty strings, and other non-numeric values are rejected (comparison returns `false`). Added tests for both `true > 0` and `null >= 0` cases.

**Comment by vignesh07** [comment]
Reviewed for relevance/consistency/focus.

This functionality is already on main (condition comparison operators).

Closing as completed to keep the PR queue clean.

- Landed commit(s): 05e34d7

Thanks @scottgl9 for the contribution.
